src/toxcore/onion/client/nodes_pool.rs

```diff
@@ -52,6 +56,47 @@ impl NodesPool {
     pub fn len(&self) -> usize {
         self.nodes.len()
     }
+
+    /// Build new random onion path with first UDP node.
+    pub fn udp_path(&self) -> Option<OnionPath> {
```

Comment:
Is there a way to shuffle a Vec and take first three items?

Reply:
Not with sodiumoxide being used for rng. Also shuffling might be less performant nodes pool is filled.